File: day1-1.rb
def captcha(num)
  string = num.to_s
  string = string + string[0]
  array = string.split ''
  sum = 0
  array.each_with_index do |str, index|
    if str == array[index + 1]
      sum += str.to_i
    end
  end
  sum
end

puts captcha 1122 # 3
puts captcha 1111 # 4
puts captcha 1234 # 0
puts captcha 91212129 # 9

File: day1-1.js
// we have to pass in the number as a string for larger numbers
function captcha(string) {
  var array = string.split('')
  var sum = 0
  array.forEach(function(str, index) {
    if (str == array[(index + 1) % array.length]) {
      sum += parseInt(str)
    }
  })
  return sum
}

console.log(captcha('1122')) // 3
console.log(captcha('1111')) // 4
console.log(captcha('1234')) // 0
console.log(captcha('91212129')) // 9
